spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/ReactiveMongoTemplate.java

```diff
@@ -826,6 +831,60 @@ public <T> Flux<T> find(@Nullable Query query, Class<T> entityClass, String coll
 				query.getFieldsObject(), entityClass, new QueryFindPublisherPreparer(query, entityClass));
 	}
 
+	@Override
+	public <T> Mono<CursorWindow<T>> findWindow(CursorRequest cursorRequest, Query query, Class<T> entityType) {
+
+		Assert.notNull(entityType, "Entity type must not be null");
+
+		return findWindow(cursorRequest, query, entityType, getCollectionName(entityType));
+	}
+
+	@Override
+	// TODO: CursorRequest and Query declare both a Sort. Which one has precedence?
+	// CursorRequest is similar to Pageable in the sense of being required to define a sort order
```

Comment:
That we have to ask ourselves this is an indicator that maybe the request should not be allowed to redefine sorting.